### DISCUSSION TITLE
Using Backpack DevTools with Existing Database Tables for Migrations, Models and CRUDs

### DISCUSSION THREAD

**Discussion by abenzakour** [Laravel-Backpack/community-forum#1171] [Q&A (Help)]
Hi everyone,

I’ve been using Backpack for Laravel to quickly generate my CRUD interfaces, and I recently purchased the **DevTools** package. However, I already have my database tables set up and would like to avoid manually recreating migrations and models through DevTools.

Is there a way to reverse-engineer the existing database schema to automatically generate the corresponding Laravel migrations and models...., and then use Backpack to build the CRUDs ? If possible, I’d love to integrate the existing database structure with Backpack's DevTools to streamline the process.

Thanks in advance for your help!

**Comment by jcastroa87** [comment]
Hello @abenzakour 

DevTools still have not this reverse-engineer features for databases, but about models is there is existing model, you dont need to create new ones, DevTools will read it.

If you have a ready scheme, here is a package who maybe can help you to create a migration from an existing table https://github.com/kitloong/laravel-migrations-generator

I hope this help.

Cheers.

**Reply by abenzakour** [reply]
Thank you @jcastroa87 , yes this package is helpful to get the migrations and make easy the CI/CD flow by only executing `php artisan migrate` instead of importing the SQL script.... 

But still need to create it manually via Devtools to generate CRUDs..

**Reply by jcastroa87** [reply]
As you ask at begging, this package will help with "Is there a way to reverse-engineer the existing database schema to automatically generate the corresponding Laravel migrations", about model i didn't know a way to reverse from table or migration, so yes, that's need to be manually.

There is a [plan to do this in a future version](https://github.com/Laravel-Backpack/community-forum/issues/222), but still not in our backlog.

Cheers.

**Comment by jcastroa87** [comment]
Due to not activity, I will close this issue, but please feel free to re-open or create a new one if needed.

Cheers.
